src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -686,181 +674,105 @@ bool ModifierManager::parse_scripts(DefinitionManager const& definition_manager)
-) const {
-	return expect_validated_modifier_value_and_default(
-		modifier_callback, type, default_callback, [](ModifierEffect const&) -> bool {
-			return true;
-		}
+NodeTools::key_value_callback_t ModifierManager::expect_unit_terrain_modifier(ModifierValue& modifier_value, TerrainType const& terrain_type) const {
+	return _expect_modifier_effect_with_fallback(terrain_modifier_effects, modifier_value, 
+	expect_base_province_modifier(modifier_value)
 	);
 }
-
-node_callback_t ModifierManager::expect_modifier_value(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type
-) const {
-	return expect_modifier_value_and_default(modifier_callback, type, key_value_invalid_callback);
-}
-
-node_callback_t ModifierManager::expect_whitelisted_modifier_value_and_default(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, string_set_t const& whitelist,
-	key_value_callback_t default_callback
-) const {
-	return expect_validated_modifier_value_and_default(
-		modifier_callback, type, default_callback,
-		[&whitelist](ModifierEffect const& effect) -> bool {
-			return whitelist.contains(effect.get_identifier());
-		}
+NodeTools::key_value_callback_t ModifierManager::expect_base_country_modifier(ModifierValue& modifier_value) const {
+	return _expect_modifier_effect_with_fallback(base_country_modifier_effects, modifier_value, 
+	_expect_shared_tech_country_modifier_effect(modifier_value)
 	);
 }
-
-node_callback_t ModifierManager::expect_whitelisted_modifier_value(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, string_set_t const& whitelist
-) const {
-	return expect_whitelisted_modifier_value_and_default(modifier_callback, type, whitelist, key_value_invalid_callback);
+NodeTools::key_value_callback_t ModifierManager::expect_base_province_modifier(ModifierValue& modifier_value) const {
+	return _expect_modifier_effect_with_fallback(base_province_modifier_effects, modifier_value, 
+	expect_base_country_modifier(modifier_value)
+	);
 }
-
-node_callback_t ModifierManager::expect_modifier_value_and_key_map_and_default(
-	callback_t<ModifierValue&&> modifier_callback, Modifier::modifier_type_t type, key_value_callback_t default_callback,
-	key_map_t&& key_map
-) const {
-	return [this, modifier_callback, type, default_callback, key_map = std::move(key_map)](
-		ast::NodeCPtr node
-	) mutable -> bool {
-		bool ret = expect_modifier_value_and_default(
-			modifier_callback, type, dictionary_keys_callback(key_map, default_callback)
-		)(node);
-
-		ret &= check_key_map_counts(key_map);
-
-		return ret;
+NodeTools::key_value_callback_t ModifierManager::expect_province_event_modifier(ModifierValue& modifier_value) const {
+	return [this, &modifier_value](auto const& key, auto const& value) {
+		if(strings_equal_case_insensitive(key, "movement_cost")) {
+			return _add_modifier_cb(modifier_value, modifier_effect_cache.movement_cost_percentage_change, value);
+		}
```

Comment:
Why is this handled specially? Is it an effect that is valid in the same way across multiple modifier types?

Reply:
This is a special case. It only has an ingame effect when applied via province event modifier. Otherwise it shows up but does nothing.